src/qc_data_preparation/helpers/dataset.py

```diff
@@ -90,3 +95,143 @@ def _exists(self) -> bool:
 
     def _describe(self) -> Dict[str, Any]:
         return dict(filepath=self._filepath)
+
+
+class PTModelDataset(AbstractDataSet):
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        self._tmp_prefix = "kedro_pytorch_tmp"  # temp prefix pattern
+        self._filepath = filepath
+        self._load_args = load_args
+        self._save_args = save_args
+        _fs_args = {}
+        protocol, path = get_protocol_and_path(filepath)
+        if protocol == "file":
+            _fs_args.setdefault("auto_mkdir", True)
+
+        self._protocol = protocol
+        self._fs = fsspec.filesystem(self._protocol)
+
+    def _load(self) -> torch.nn.Module:
+        state_dict = torch.load(self._filepath)
+        model = PT_Autoencoder(**self._load_args)
+        model.load_state_dict(state_dict)
+        return model
+
+    def _save(self, model: torch.nn.Module) -> None:
+        save_path = self._filepath
+        with tempfile.TemporaryDirectory(prefix=self._tmp_prefix) as path:
+            file_path = path + "/" + os.path.basename(save_path)
+            torch.save(model.state_dict(), file_path, **self._save_args)
+            if self._fs.exists(save_path):
+                self._fs.rm(save_path, recursive=True)
+            self._fs.put(file_path, save_path, recursive=False)
+
+    def _exists(self) -> bool:
+        return Path(self._filepath.as_posix()).exists()
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            filepath=self._filepath,
+            model=PT_Autoencoder,
+            load_args=self._load_args,
+            save_args=self._save_args,
+        )
+
+
+class TFPTModelDataset(AbstractDataSet):
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        self._save_args = {} if not save_args else save_args
+        self._load_args = {} if not load_args else load_args
+        self._filepath = filepath
+        self._model_io = None
+
+    def _load(self) -> "torch.nn.Module | tf.keras.Model":
+        """
+        This method tries to load one of the supported models,
+        :py:class::`~.PT_Autoencoder`, or :py:class::`~.TF_Autoencoder`.
+        """
+        if self._model_io:
+            return self._model_io._load()
+        else:  # checks are potentially dangerous since no specific error is raised
+            try:
+                self._init_tf_io()
+                return self._model_io._load()
+            except IsADirectoryError:
+                pass
+            try:
+                self._init_pt_io()
+                return self._model_io._load()
+            except IsADirectoryError:
+                pass
+            raise TypeError(f"Type of model in {self._filepath} is not supported")
```

Comment:
Since both models are used the same, perhaps a loop over both is clearer. That's assuming they will always fail the same way, though.
```suggestion
        for model_init in (self._init_tf_io, self._init_pt_io):
            try:
                model_init()
                return self._model_io._load()
            except IsADirectoryError:
                pass
            raise TypeError(f"Type of model in {self._filepath} is not supported")
```